qlog/qlog.go

```diff
@@ -17,6 +18,27 @@ import (
 	"github.com/francoispqt/gojay"
 )
 
+// Setting of this only works when quic-go is used as a library.
+// When building a binary from this repository, the version can be set using the following go build flag:
+// -ldflags="-X github.com/lucas-clemente/quic-go/qlog.quicGoVersion=foobar"
+var quicGoVersion = "(devel)"
+
+func init() {
+	info, ok := debug.ReadBuildInfo()
+	if !ok { // no build info available. This happens when quic-go is not used as a library.
+		return
+	}
+	for _, d := range info.Deps {
```

Comment:
you could also check the case where quic-go is the main module, and has a version.

this will be rather niche, but it could happen with e.g. `go test github.com/lucas-clemente/quic-go/...` from another module.

Reply:
I'm not sure if that's a case that we really need to cover.